src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -367,475 +326,520 @@ private void AnalyzeNodeForDtdProcessingOverloads(SyntaxNodeAnalysisContext cont
-                        if (lhsExpressionNode == null)
-                        {
-                            return;
-                        }
+                // if the XmlResolver or Dtdprocessing property is explicitly set when created, and is to an insecure value, generate a warning
+                if ((env.IsXmlResolverSet && !env.IsSecureResolver) ||
+                    (env.IsDtdProcessingSet && !env.IsDtdProcessingDisabled))
+                {
+                    Diagnostic diag = Diagnostic.Create(
+                        RuleDoNotUseInsecureDTDProcessing,
+                        env.XmlTextReaderDefinition.GetLocation(),
+                        SecurityDiagnosticHelpers.GetLocalizableResourceString(
+                            nameof(DesktopAnalyzersResources.XmlTextReaderSetInsecureResolutionMessage)
+                        )
+                    );
+                    context.ReportDiagnostic(diag);
+                }
+                // if the XmlResolver or Dtdprocessing property is not explicitly set when constructed for a non-temp XmlTextReader object, add env to the dictionary.
+                else if (variable != null && !(env.IsDtdProcessingSet && env.IsXmlResolverSet))
+                {
+                    _xmlTextReaderEnvironments[variable] = env;
+                }
+                // if the is not set or set to Parse for a temporary object, report right now.
+                else if (variable == null && !(env.IsDtdProcessingSet && env.IsXmlResolverSet && env.IsDtdProcessingDisabled && env.IsSecureResolver))
+                {
+                    Diagnostic diag = Diagnostic.Create(
+                        RuleDoNotUseInsecureDTDProcessing,
+                        env.XmlTextReaderDefinition.GetLocation(),
+                        SecurityDiagnosticHelpers.GetLocalizableResourceString(
+                            nameof(DesktopAnalyzersResources.XmlTextReaderConstructedWithNoSecureResolutionMessage)
+                        )
+                    );
+                    context.ReportDiagnostic(diag);
+                }
+            }
 
-                        ISymbol lhsExpressionSymbol = SyntaxNodeHelper.GetSymbol(lhsExpressionNode, model);
-                        if (lhsExpressionSymbol == null)
-                        {
-                            return;
-                        }
+            private void AnalyzeObjectCreationForXmlReaderSettings(OperationAnalysisContext context, ISymbol variable, IObjectCreationExpression objCreation)
+            {
+                XmlReaderSettingsEnvironment xmlReaderSettingsEnv = new XmlReaderSettingsEnvironment(_isFrameworkSecure);
 
-                        XmlTextReaderEnvironment env = null;
-                        _xmlTextReaderEnvironments.TryGetValue(lhsExpressionSymbol, out env);
+                if (variable != null)
+                {
+                    _xmlReaderSettingsEnvironments[variable] = xmlReaderSettingsEnv;
```

Comment:
Well, it should be, if a XmlReaderSetting is created or declared, and captured by this analyzer, it should have an item in `_xmlReaderSettingsEnvironments`. If it is just a method parameter, no. I am not sure I answered the question.